databroker/headersource/core.py

```diff
@@ -515,7 +515,12 @@ def insert_run_start(run_start_col, run_start_cache,
     col = run_start_col
     run_start = dict(time=time, uid=uid, **kwargs)
     apply_to_dict_recursively(run_start, sanitize_np)
-
+    try:
+        if run_start_given_uid(uid, run_start_col,
```

Comment:
The reasoning here is to address the issue in one place for all backends. The "brute-force" check is exactly the same operation the database will perform when it is enforcing uniqueness.